include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -216,28 +218,67 @@ struct __parallel_merge_submitter<_IdType, __internal::__optional_kernel_name<_N
 
         const _IdType __steps = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __chunk);
 
-        auto __event = __exec.queue().submit(
-            [&__rng1, &__rng2, &__rng3, __comp, __chunk, __steps, __n1, __n2](sycl::handler& __cgh) {
-                oneapi::dpl::__ranges::__require_access(__cgh, __rng1, __rng2, __rng3);
-                __cgh.parallel_for<_Name...>(sycl::range</*dim=*/1>(__steps), [=](sycl::item</*dim=*/1> __item_id) {
-                    const _IdType __i_elem = __item_id.get_linear_id() * __chunk;
-                    const auto __start =
-                        __find_start_point(__rng1, _IdType{0}, __n1, __rng2, _IdType{0}, __n2, __i_elem, __comp);
-                    __serial_merge(__rng1, __rng2, __rng3, __start.first, __start.second, __i_elem, __chunk, __n1, __n2,
-                                   __comp);
-                });
+        using __val_t = _split_point_t<_IdType>;
+        using __result_and_scratch_storage_t = __result_and_scratch_storage<_ExecutionPolicy, __val_t>;
+        __result_and_scratch_storage_t* __p_res_storage = nullptr;
+
+        if constexpr (_OutSizeLimit{})
+            __p_res_storage = new __result_and_scratch_storage_t(__exec, 1, 0);
+        else
+        {
+            assert(__rng3.size() >= __n1 + __n2);
+        }
```

Comment:
```suggestion
            assert(__rng3.size() >= __n1 + __n2);
```

Reply:
I consciously kept `{}`, because `assert` is a macro, which may defined to nothing.

Reply:
I'm sure there's no reason to worry in this case.